kotlin_spark/src/main/java/com/sleazyweasel/eboshi/AccountRoutes.kt

```diff
@@ -0,0 +1,49 @@
+package com.sleazyweasel.eboshi
+
+import org.eclipse.jetty.http.HttpStatus
+import org.jose4j.jwt.consumer.InvalidJwtException
+import spark.Request
+import spark.Response
+import javax.inject.Inject
+
+class AccountRoutes @Inject constructor(accountDataAccess: AccountDataAccess, auth: Auth) {
+    val create = { input: JsonApiRequest, response: Response ->
+        val submittedUserAttributes = input.data.attributes
+        val user = Account(submittedUserAttributes)
+        val (hashed, salt) = auth.encrypt(user.password!!)
```

Comment:
actually, in real life I think we'd check for a null password before trying to encrypt it, and return the 401 then, rather than using exception handling for the control flow.